homeassistant/data_entry_flow.py

```diff
@@ -219,6 +220,7 @@ def async_create_entry(
             "data": data,
             "description": description,
             "description_placeholders": description_placeholders,
+            **kwargs,
```

Comment:
I am not a fan because it feels weird. Data should be the data for the config entry, all other is metadata about the step. 

Not sure what the correct thing is .

Reply:
Whatever way that makes it possible to set disable_new_entities to True for a newly created config entry works for me. I wouldn't want to handle disabling the entities manually, and it would be weird if the UI shows new entities being enabled by default where in fact they are created disabled.

Anyway, not sure I understand your concern, but I do think this (options and system_options) _is_ stuff about the config entry, it's not a step thing. It gets persistent with the config entry, UI shows the new entity enablement toggle for the entry, etc.

Or maybe you're concerned about opening the big kwargs possibility to data_entry_flow? As mentioned, that could be left alone and replaced by doing this in ConfigFlow just for options and system_options.